classes/notification/managerDelegate/QueryNotificationManager.inc.php

```diff
@@ -101,8 +101,8 @@ public function getNotificationUrl($request, $notification) {
 		assert(is_a($query, 'Query'));
 		$submission = $this->getQuerySubmission($query);
 
-		import('lib.pkp.controllers.grid.submissions.SubmissionsListGridCellProvider');
-		return SubmissionsListGridCellProvider::getUrlByUserRoles($request, $submission, $notification->getUserId());
+		import('classes.core.ServicesContainer');
+		return ServicesContainer::instance()->getWorklowUrlByUserRoles($submission, $notification->getUserId());
```

Comment:
I believe this should be `ServicesContainer::instance()->get('submission')->getWorklowUrlByUserRoles()`